File: src/#897/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}

var increasingBST = function (root) {
  var arr = []
  inOrder(root)
  var head = new TreeNode(arr[0])
  var current = head

  for (let i = 1; i < arr.length; i++) {
    let node = new TreeNode(arr[i])
    current.right = node
    current = current.right
  }

  return head
  function inOrder(root) {
    if (root === null) {
      return
    }
    inOrder(root.left)
    arr.push(root.val)
    inOrder(root.right)
  }
}

File: src/#897/solution.go
package solution

type TreeNode struct {
	Val         int
	Left, Right *TreeNode
}

func increasingBST(root *TreeNode) *TreeNode {
	var arr []int
	inOrder(root, &arr)

	head := &TreeNode{Val: arr[0]}
	current := head
	for _, v := range arr[1:] {
		current.Right = &TreeNode{Val: v}
		current = current.Right
	}

	return head
}

func inOrder(root *TreeNode, arr *[]int) {
	if root == nil {
		return
	}

	inOrder(root.Left, arr)
	*arr = append(*arr, root.Val)
	inOrder(root.Right, arr)
}
